### ISSUE TITLE
Json bound property converter?

### ISSUE THREAD

**Issue by careri** [Starcounter/Starcounter#2949]
Are there any plans for supporting converters when binding to a value in Json.
Example (using attribute from Newtonsoft.Json):
```
class Data { 
   [JsonConverter(typeof(VersionTypeConverter))]
   public Version MyVersion { get; set; }
}
```

Converter:
```
class VersionTypeConverter : JsonConverter
{
    public override bool CanConvert(Type objectType)
    {
        return typeof(Version).Equals(objectType);
    }

    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
    {
        return Version.Parse((string)reader.Value);
    }

    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
    {
        serializer.Serialize(writer, value.ToString());
    }
}
```

Json:
```
{
   "MyVersion" : "1.2.3.4"
}
```

**Comment by chrhol** [comment]
The short answer is no, there is no current plans to support converters on TypedJson. The current implementation is mainly focused on binding to database data, where these kind of values are not supported.

However, we are aiming at making the TypedJson implementation more independent from Starcounter specifics so this sounds like a nice feature to add. We will definitely have this in consideration.

**Comment by chrhol** [comment]
Of course, you can get the same behaviour by binding the json-property to a codebehind property that does the exact same thing as the converter in your example so maybe a specific Converter implementation is not needed.

The only drawback with this solution is that you need to specify the binding-path manually since properties in JSON-by-example file and codebehind cannot have the same name currently. 

There should be an issue about using the codebehind-property directly if specified in both JSON and codebehind but I cannot find it at the moment.

**Comment by chrhol** [comment]
Hmm, I cannot find it even after browsing through all TypedJSON issues, maybe it's not labeled correctly. Anyways I created a new issue, #2964

**Comment by careri** [comment]
Yepp, but it's much cleaner/simpler to just tag the properties with the attributes like in the example
